### DISCUSSION TITLE
heart rate monitoring via audioplethysmography

### DISCUSSION THREAD

**Discussion by khimaros** [pine64/OpenPineBuds#71] [Ideas]
pie in the sky idea. would be the first consumer device capable of doing this. as discussed here: https://9to5google.com/2023/10/27/google-anc-earbuds-heart-rate/

and the full research paper available here: https://storage.googleapis.com/pub-tools-public-publication-data/pdf/e107049945d4e506308b995a7d9d77daff1f6133.pdf

**Comment by shymega** [comment]
Whilst interesting, I think it's outside of the scope of the firmware. And the clients of the data measured by the PineBuds probably would be limited. Just my thoughts...

**Reply by khimaros** [reply]
@shymega could you elaborate on this a bit? how would the clients of the data be limited? do you mean the hardware is too resource constrained?

**Comment by amazonparrot** [comment]
I think this idea is actually very good, but I have yet to find out how to access each mics individually, if I know how to access the mics individually I think this would be doable. There isn't much commercial earbuds that can do this, if this application is included this would definite cause some attentions. @khimaros please let me know if you know anyone is currently working on this?

**Reply by khimaros** [reply]
i'm not aware of anyone working on this. i would be open to collaborating. if you want someone to bounce ideas, let me know.

**Reply by khimaros** [reply]
@amazonparrot as an interesting starting point, this is the branch where someone started working on ANC support: https://github.com/pine64/OpenPineBuds/compare/main...anc-testing

if you look at the `tgt_hardware` source files, you can see where they are configuring the feed forward and feed back microphones for the left and right earbud. does that help point you in the right direction?

**Reply by khimaros** [reply]
maybe @Ralim could provide some context on that branch.

**Reply by Ralim** [reply]
I'm not sure what context your asking for here?
I configured the mics to match the schematics for how the unit is wired up. Then I ported the tuning data from a dump of my units.

It still doesn't work properly but I haven't looked into it further as I wasn't sure where to look next tbqh.

**Reply by khimaros** [reply]
@Ralim as a baseline, do you believe you were getting accurate signal from each of the channels as you configured them and they were correctly mapped?

**Reply by Ralim** [reply]
Correctly mapped: yes
Accurate data: I think so, though lightly suspect data is coming in quiet.

**Comment by matvarg** [comment]
Any progress on this interesting topic?
